cl/scrapers/utils.py

```diff
@@ -446,3 +449,49 @@ def scraped_citation_object_is_valid(citation_object: dict) -> bool:
         logger.error("Parsed reporter '%s' does not exist", parsed_reporter)
 
     return False
+
+
+def save_response(site: AbstractSite) -> None:
+    """Stores scrapers responses content and headers in a S3 bucket
+
+    This is passed to juriscraper's Site objects as the
+    `save_response_fn` argument, which will make Juriscraper
+    save every response
+
+    :param site: the Site object, used to access the saved response
+    :return None
+    """
+
+    storage = S3GlacierInstantRetrievalStorage()
+    response = site.request["response"]
+    headers = dict(response.headers)
+
+    scraper_id = site.court_id.split(".")[-1]
+    scrape_type = site.court_id.split(".")[1]  # opinions or oral args
+    now_str = datetime.now().strftime("%Y_%m/%d/%H_%M_%S")
+    base_name = f"responses/{scrape_type}/{scraper_id}/{now_str}"
+
+    try:
+        # both tests and parses JSON content
+        content = json.loads(response.content)
+        extension = "json"
+    except:
+        content = response.content
+        extension = "html"
+
+    # Append the headers to the response object, if it's a JSON
+    if isinstance(content, list):
+        content.append({"response_headers": headers})
+        content = json.dumps(content, indent=4)
+    elif isinstance(content, dict):
+        content["response_headers"] = headers
+        content = json.dumps(content, indent=4)
+    else:
+        # Appending the response headers to the content would save a
+        # PUT request and an extra download step, but it may cause
+        # encoding and upload conflicts, so we save them apart
+        json_string = json.dumps(dict(response.headers), indent=4)
+        storage.save(f"{base_name}_headers.json", ContentFile(json_string))
```

Comment:
If you just have a rule of always uploading two files, you could really simplify this, right? Just send response.content in file number 1 and response.headers in file number 2?

Feels a bit tidier in a way too, since you won't be adding headers to a scraped JSON object?